newIDE/app/src/Profile/EditProfileDialog.js

```diff
@@ -0,0 +1,100 @@
+// @flow
+import { Trans } from '@lingui/macro';
+
+import React, { Component } from 'react';
+import FlatButton from '../UI/FlatButton';
+import RaisedButton from '../UI/RaisedButton';
+import Dialog from '../UI/Dialog';
+import TextField from '../UI/TextField';
+import {
+  type EditForm,
+  type AuthError,
+  type Profile,
+} from '../Utils/GDevelopServices/Authentication';
+import LeftLoader from '../UI/LeftLoader';
+import { ColumnStackLayout } from '../UI/Layout';
+
+type Props = {|
+  profile: Profile,
+  onClose: () => void,
+  onEdit: (form: EditForm) => void,
+  editInProgress: boolean,
+  error: ?AuthError,
+|};
+
+type State = {|
+  form: EditForm,
+|};
+
+export const getUsernameErrorText = (error: ?AuthError) => {
+  if (!error) return undefined;
+
+  if (error.code === 'auth/username-used')
+    return 'This username is already used, please pick another one.';
+  return undefined;
+};
+
+export default class EditDialog extends Component<Props, State> {
```

Comment:
nitpicking: we could have gone for a functional component with hooks - won't change a lot here but seems like a good default choice to use hooks rather than classes going forward.

Reply:
I'll keep it in mind moving forward, but yes I agree with you!